shared/src/simulation/activities/simulation-activity.ts

```diff
@@ -0,0 +1,23 @@
+import type { SimulatedRegion } from '../../models';
+import type { ExerciseState } from '../../state';
+import type { Constructor, Mutable, UUID } from '../../utils';
+
+export class SimulationActivityState {
+    readonly type!: `${string}Activity`;
+    readonly id!: UUID;
+}
+
+export interface SimulationActivity<S extends SimulationActivityState> {
+    readonly activityState: Constructor<S>;
+    readonly tick: (
+        draftState: Mutable<ExerciseState>,
+        simulatedRegion: Mutable<SimulatedRegion>,
+        activityState: Mutable<S>,
+        tickInterval: number
+    ) => void;
+    readonly terminate?: (
+        draftState: Mutable<ExerciseState>,
+        simulatedRegion: Mutable<SimulatedRegion>,
+        activityId: UUID
+    ) => void;
```

Comment:
I would rename this to `onTerminate` (or similar) to state that this is just an event handler and you have to use `terminateActivity()` (from the utils) if you want to terminate an activity.